src/components/calcite-ui-icon/calcite-ui-icon.tsx

```diff
@@ -0,0 +1,172 @@
+import { Component, h, Element, Prop, State, Watch, Host } from "@stencil/core";
+import { CSS } from "./resources";
+import { getElementDir } from "../../utils/dom";
+import { fetchIcon, scaleToPx } from "./utils";
+import { Scale } from "../../interfaces/UIIcon";
+import { Theme } from "../../interfaces/common";
+
+@Component({
+  assetsDirs: ["assets"],
+  tag: "calcite-ui-icon",
+  styleUrl: "calcite-ui-icon.scss",
+  shadow: true
+})
+export class CalciteUIIcon {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element()
+  el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, the icon will be filled.
+   */
+  @Prop({
+    reflect: true,
+    mutable: true
+  })
+  filled: boolean = false;
+
+  /**
+   * The UI icon to display. Name must match icon from https://esri.github.io/calcite-ui-icons/.
```

Comment:
Should we say `Must match icon name...`?

Mention that its the name and not the src or anything.